src/scenes/Initial.js

```diff
@@ -1,42 +1,41 @@
 import React, {Component, PropTypes} from 'react';
 import {AsyncStorage, Image, StyleSheet, View, Text} from 'react-native';
-import store from '../store'
 import {connect} from 'react-redux'
+import {fetchRegionFromStorage} from '../actions/region';
+import {fetchDirectionFromStorage} from '../actions/direction';
+import {fetchLanguageFromStorage} from '../actions/language';
+import {fetchCountryFromStorage} from '../actions/country';
 
-import { fetchRegionFromStorage } from '../actions/region';
-import { fetchDirectionFromStorage } from '../actions/direction';
-import { fetchLanguageFromStorage } from '../actions/language';
-import { fetchCountryFromStorage } from '../actions/country';
-
- class Initial extends Component {
+class Initial extends Component {
 
     static contextTypes = {
+        drawer: PropTypes.object.isRequired,
         navigator: PropTypes.object.isRequired
     };
 
-  async componentDidMount() {
-    const { dispatch } = this.props;
-    const r = await AsyncStorage.getItem('regionCache');
-
-    await dispatch(fetchRegionFromStorage());
-    await dispatch(fetchDirectionFromStorage());
-    await dispatch(fetchLanguageFromStorage());
-    await dispatch(fetchCountryFromStorage());
-
-    if(!(r||false)) {
-      this.context.navigator.to('countryChoice');
-    } else {
-      this.context.navigator.to('info');
+    async componentDidMount() {
+        const {navigator} = this.context;
+        const {dispatch} = this.props;
+        const region = await AsyncStorage.getItem('regionCache');
+
+        await dispatch(fetchRegionFromStorage());
+        await dispatch(fetchDirectionFromStorage());
+        await dispatch(fetchLanguageFromStorage());
+        await dispatch(fetchCountryFromStorage());
```

Comment:
Do we have to care about performance here? If so you can use Promise.all (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) or AsyncStorage multi-get (https://facebook.github.io/react-native/docs/asyncstorage.html#multiget)